src/adagrams.js

```diff
@@ -1,15 +1,127 @@
-export const drawLetters = () => {
-  // Implement this method for wave 1
+/* eslint-disable quotes */
+/* eslint-disable no-unused-vars */
+
+const LETTER_POOL = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
 };
 
+// Implement this method for wave 1
+export const drawLetters = () => {
+  const copyOfLetters = JSON.parse(JSON.stringify(LETTER_POOL))
+  let tenLetters = [];
+  while (tenLetters.length < 10) {
+    const keys = Object.keys(copyOfLetters)
+    let randomLetter = keys[Math.floor(Math.random() * keys.length)];
+    
+    if (copyOfLetters[randomLetter] > 0) {
+      tenLetters.push(randomLetter); 
+      copyOfLetters[randomLetter] -= 1;
+    }
+  }
+  return tenLetters;
+}
+
+
+// Implement this method for wave 2
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  let copyLettersInHand = lettersInHand;
+  for (let letter of input) {
+    if (copyLettersInHand.includes(letter)) {
+      copyLettersInHand.splice(input[letter],1);
```

Comment:
This works in terms of passing our tests, but consider an input like `'oof'`. If you had a hand with `['f', 'o', 'o']`, then the function should return `true`, but yours returns `false`. That's because you're always splicing the first element of the array, due to using `input[letter]` as the first argument. That actually returns `undefined`, because you're trying to index a string by a substring, e.g. `cat['a']`, which will always be undefined (strings can be indexed by positions, i.e. numbers, such as `cat[1` which would return `'a'`. But you also shouldn't be passing `undefined` as the first argument to the `splice()` call (I didn't actually realize that you _could_ pass `undefined` to the first argument of `splice()`. but evidently, it treats it as a `0` because in testing this, I found that calling `.splice(undefined, 1)` on an array always returned the first element. So, what you actually want here is an index, specifically the position of the letter in the array, so if you did:
```js
      const idx = copyLettersInHand.indexOf(letter);
      copyLettersInHand.splice(idx, 1);
```
that would cover the case I mentioned above.